go.mod

```diff
@@ -1,4 +1,4 @@
-module github.com/bwmarrin/discordgo
```

Comment:
We can't make this change for sure :)  This should not be part of your PR.

Reply:
Oh yeah, sorry about that. I just needed to commit this so that I could use the code when I import the module in my go.mod for my other repo that this code was using. We can remove that part of the PR definitely.

Reply:
A small tip, you could use `replace` directive in `go.mod` to get the same result, but without changing the target module 😄

Reply:
Yes, I am sorry for not thinking about this. 

> First of all, thanks for making the PR!
> 
> Would like to share some of my own thoughts on the use case you have described, if you don't mind:
> 
> > I want to be able to feed message data into the model exactly as they would appear coming from discord
> 
> You can hook up a raw event handler — `func(s *discordgo.Session, e *discordgo.Event)`, from which you would be able to access `RawData` of the event:
> 
> ```go
> s.AddHandler(func(s *discordgo.Session, e *discordgo.Event) {
>    if e.Type != "MESSAGE_CREATE" {
>         return
>    }
>    fmt.Println(e.RawData)
> })
> ```
> 
> > Calling json.Marshal() on the message object does not give the same output as the json received from the Discord API which is why I wanted a way to store the raw response.
> 
> That's by design, this data structure was meant only to be received and read. It will differ from the data sent by Discord due to it being partial sometimes.

Thanks for the feedback. OK so the solution you are proposing would only work for websocket connections, correct? There is not a way to integrate this with the REST API, for example to marshal the []*discordgo.Message received from session.ChannelMessages(channelID string, limit int, beforeID, afterID, aroundID string)?